bin/check-azurerm-monitor-metric.rb

```diff
@@ -353,4 +364,83 @@ def timespan
   def handle_response(res)
     critical "Failed to get metric:\n#{res.body}" if res.code.to_i >= 300
   end
+
+  def verify_results_with_aggregation
+    request_values = extract_request_values
+
+    aggregated_value = aggregate_request_values(request_values, config[:aggregate_results])
+
+    error_type = verify_result(aggregated_value)
+
+    return_error_message(error_type, request_values[0][:name], aggregated_value)
+  end
+
+  def extract_request_values
+    values = []
+    metric_response[:value].each do |metric_resp_value|
+      name = metric_resp_value[:name] ? metric_resp_value[:name][:value] : ''
+
+      next if metric_resp_value[:timeseries].empty?
+
+      metric_resp_value[:timeseries].each do |ts|
+        ts[:data].each do |metric_value|
+          if metric_value[metric_value_key]
+            values << {
+              value: metric_value[metric_value_key].to_f,
+              name: name
+            }
+          end
+        end
+      end
+    end
+    values
+  end
+
+  def aggregate_request_values(request_values, aggregation_type)
+    result_values = []
+
+    request_values.each do |metric_val|
+      result_values.push(metric_val[:value])
+    end
+
+    case aggregation_type
+    when 'average'
+      result_value = result_values.inject { |sum, el| sum + el }.to_f / result_values.size
```

Comment:
Is `el` short for `element`? If so I would suggest using the full spelling or another more descriptive name.